2022-08-02 23:45:19,055 ERROR: Exception on /login [GET] [in d:\github\microblog2\lib\site-packages\flask\app.py:1455]
Traceback (most recent call last):
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "D:\microblog2\app\routes.py", line 72, in login
    return render_template('login.html', title='Sign In', form=form)
  File "d:\github\microblog2\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "d:\github\microblog2\lib\site-packages\jinja2\loaders.py", line 138, in load
    code = environment.compile(source, name, filename)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 768, in compile
    self.handle_exception(source=source_hint)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\microblog2\app\templates\login.html", line 26, in template
    <a href="{{ url_for{'reset_password_request') }}">Click to Reset It</a>
jinja2.exceptions.TemplateSyntaxError: expected token 'end of print statement', got '{'
2022-08-02 23:45:20,112 ERROR: Exception on /login [GET] [in d:\github\microblog2\lib\site-packages\flask\app.py:1455]
Traceback (most recent call last):
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "D:\microblog2\app\routes.py", line 72, in login
    return render_template('login.html', title='Sign In', form=form)
  File "d:\github\microblog2\lib\site-packages\flask\templating.py", line 149, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 1081, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 1010, in get_template
    return self._load_template(name, globals)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 969, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "d:\github\microblog2\lib\site-packages\jinja2\loaders.py", line 138, in load
    code = environment.compile(source, name, filename)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 768, in compile
    self.handle_exception(source=source_hint)
  File "d:\github\microblog2\lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "D:\microblog2\app\templates\login.html", line 26, in template
    <a href="{{ url_for{'reset_password_request') }}">Click to Reset It</a>
jinja2.exceptions.TemplateSyntaxError: expected token 'end of print statement', got '{'
2022-08-02 23:46:35,611 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:46]
2022-08-02 23:47:25,715 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:46]
2022-08-02 23:49:26,929 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:46]
2022-08-02 23:49:55,361 ERROR: Exception on /reset_password_request [POST] [in d:\github\microblog2\lib\site-packages\flask\app.py:1455]
Traceback (most recent call last):
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\github\microblog2\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "D:\microblog2\app\routes.py", line 103, in reset_password_request
    send_password_reset_email(user)
  File "D:\microblog2\app\email.py", line 20, in send_password_reset_email
    token = user.get_reset_password_token()
  File "D:\microblog2\app\models.py", line 69, in get_reset_password_token
    app.config['SECRET_KEY'], algorithm='HS256').decode('utf-8')
AttributeError: module 'app' has no attribute 'config'
2022-08-02 23:57:59,764 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:46]
2022-08-03 00:12:08,204 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:46]
2022-08-03 00:15:54,232 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:46]
2022-09-02 15:35:47,208 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 22:27:32,204 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 22:45:06,069 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:13:49,639 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:14:54,923 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:20:33,694 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:20:41,038 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:22:38,532 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:23:02,569 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:23:55,693 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:24:09,005 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:24:33,030 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:24:42,413 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:32:18,877 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:32:39,513 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-03 23:35:07,113 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:04:41,066 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:04:53,063 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:05:34,891 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:11:33,673 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:12:07,711 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:16:09,548 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
2022-09-04 00:16:30,370 INFO: Microblog Startup [in D:\microblog2\app\__init__.py:58]
